draco/analysis/transform.py

```diff
@@ -513,6 +522,19 @@ def process(self, sstream: containers.SiderealContainer) -> containers.MContaine
         # Assign the weights into the container
         ma.weight[:] = weight_sum[np.newaxis, np.newaxis, :, :]
 
+        # Divide out the m-mode sinc-suppression caused by the rectangular integration window
+        if self.remove_integration_window:
+
+            m = np.arange(mmax + 1)
+            w = np.sinc(m / nra)
+            inv_w = tools.invert_no_zero(w)
+
+            sl_vis = (slice(None),) + (np.newaxis,) * (len(ma.vis.shape) - 1)
+            ma.vis[:] *= inv_w[sl_vis]
+
+            sl_weight = (slice(None),) + (np.newaxis,) * (len(ma.vis.shape) - 1)
```

Comment:
I think you want the shape of the weight dataset here (this doesn't matter for `MModes`, but makes a difference for `HybridVisMModes`):
```suggestion
            sl_weight = (slice(None),) + (np.newaxis,) * (len(ma.weight.shape) - 1)
```